keras_cv/layers/preprocessing/fmix.py

```diff
@@ -0,0 +1,221 @@
+            tf.math.maximum(freqs, tf.convert_to_tensor([1 / tf.reduce_max([w, h])]))
+            ** decay_power,
+            tf.float32,
+        )
+
+        param_size = tf.concat(
+            [tf.constant([channel]), tf.shape(freqs), tf.constant([2])], 0
+        )
+        param = self._random_generator.random_normal(param_size)
+
+        scale = tf.expand_dims(scale, -1)[None, :]
+
+        return scale * param
+
+    def sample_mask_from_transform(self, decay, shape, ch=1):
+        # Sampling low frequency map from fourier transform.
+
+        freqs = self.apply_fftfreq(shape[0], shape[1])
+        spectrum = self.get_spectrum(freqs, decay, ch, shape[0], shape[1])
+        spectrum = tf.complex(spectrum[:, 0], spectrum[:, 1])
+
+        mask = tf.math.real(tf.signal.irfft2d(spectrum, shape))
+        mask = mask[:1, : shape[0], : shape[1]]
+
+        mask = mask - tf.reduce_min(mask)
+        mask = mask / tf.reduce_max(mask)
+        return mask
+
+    def binarise_mask(self, mask, lam, in_shape, edge_softening=0.0):
+        # Create the final mask from the sampled values.
+
+        idx = tf.argsort(tf.reshape(mask, [-1]), direction="DESCENDING")
+        mask = tf.reshape(mask, [-1])
+        num = tf.math.round(lam * tf.cast(tf.size(mask), tf.float32))
+
+        eff_soft = edge_softening
+
+        max_compare = tf.math.maximum(lam, (1 - lam))
+        min_compare = tf.math.minimum(lam, (1 - lam))
+        eff_soft = tf.where(edge_softening > max_compare, max_compare, edge_softening)
+        eff_soft = tf.where(edge_softening > min_compare, min_compare, edge_softening)
+
+        soft = tf.cast(tf.size(mask), tf.float32) * eff_soft
+        num_low = tf.cast(num - soft, tf.int32)
+        num_high = tf.cast(num + soft, tf.int32)
+
+        updates = tf.concat(
+            [
+                tf.ones((num_low,), tf.float32),
+                tf.cast(tf.linspace(1, 0, num_high - num_low), tf.float32),
+                tf.zeros((tf.size(mask) - num_high,), tf.float32),
+            ],
+            0,
+        )
+
+        mask = tf.scatter_nd(
+            tf.expand_dims(idx, -1), updates, tf.expand_dims(tf.size(mask), -1)
+        )
+
+        mask = tf.reshape(mask, in_shape)
+        return mask
+
+    def _batch_augment(self, inputs):
+        images = inputs.get("images", None)
+        labels = inputs.get("labels", None)
+        if images is None or labels is None:
+            raise ValueError(
+                "FMix expects inputs in a dictionary with format "
+                '{"images": images, "labels": labels}.'
+                f"Got: inputs = {inputs}"
+            )
+        images, lambda_sample, permutation_order = self._fmix(images)
+        if labels is not None:
+            labels = self._update_labels(labels, lambda_sample, permutation_order)
+            inputs["labels"] = labels
+        inputs["images"] = images
+        return inputs
+
+    def _augment(self, inputs):
```

Comment:
Is this the only layer in KLPs where we have this behaviour?

Reply:
`MixUp` is implemented in a similar way.

Reply:
Oh yes it is seems that it was introduced previously also in `MIxUp` by @qlzh727 with https://github.com/keras-team/keras-cv/commit/2c42c48bc6a449799fa008f1a8a808ee1952c75c